### PR TITLE
ISPN-7532 Do not log configuration warnings for templates

### PR DESCRIPTION
**Pull request by tristantarrant**
- Do not log configuration warnings for templates
- Also extract some configuration exceptions to log
- Do not disable expiration reaper in server

https://issues.jboss.org/browse/ISPN-7532

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/logging/Log.java [hunk 1]
```diff
@@ -1518,5 +1518,32 @@ TimeoutException timeoutWaitingForView(int expectedViewId, int currentViewId,
    void errorSendingResponse(ReplicableCommand command);
 
    @Message(value = "Unsupported async cache mode '%s' for transactional caches", id = 441)
-   CacheConfigurationException unsupportedAsyncCacheMode(CacheMode cacheModed);
+   CacheConfigurationException unsupportedAsyncCacheMode(CacheMode cacheMode);
+
+   @Message("Invalid cache loader configuration for '%s'.  If a cache loader is configured as a singleton, the cache loader cannot be shared in a cluster!", id = 442)
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/util/logging/Log.java]: Missing `value=`.

Comment (tristantarrant) [inline core/src/main/java/org/infinispan/util/logging/Log.java]: It seems I can omit it only if it's the only attribute

**Comment by tristantarrant** [comment]
Fixed it

**Comment by danberindei** [comment]
Seeing test failures now:

```
testNegativeTimeout(org.infinispan.xsite.XSiteStateTransferFileParsingTest)  Time elapsed: 0.022 sec  <<< FAILURE!
org.testng.TestException: 
The exception was thrown with the wrong message: expected "Timeout must be higher or equals than 1 \(one\)." but got "ISPN000449: XSite state transfer timeout must be higher or equals than 1 (one)."
```

**Comment by tristantarrant** [comment]
Fixed the test.

**Pull request merged**